src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/MarshalTests.cs

```diff
@@ -246,6 +246,23 @@ private static void VerifyBytes(IntPtr p, int length)
         }
 
         [Fact]
+        public static void ValidateExceptionForHRAPI()
+        {
+            int errorCode = -2146231029;
+            COMException getHRException = Marshal.GetExceptionForHR(errorCode) as COMException;
+            Assert.NotNull(getHRException);
+            try
+            {
+                Marshal.ThrowExceptionForHR(errorCode);
+            }
+            catch (COMException e)
+            {             
+                Assert.Equal(e.HResult, errorCode);
+                Assert.Equal(e.HResult, getHRException.HResult);
+            }
```

Comment:
I would expect the test to fail if it's anything other than ComException , what's the point in an additional catch .. Also the xunit Assert mess up the callstack  ( esply for UAPAOT)